ibis_substrait/compiler/translate.py

```diff
@@ -822,6 +822,9 @@ def selection(
             elif output_mapping := rel.common.emit.output_mapping:
                 mapping_counter = itertools.count(len(output_mapping))
                 break
+            elif relname in ("project", "filter"):
+                mapping_counter = itertools.count(len(op.table.op().schema))
+                break
         else:
             source_tables = _find_parent_tables(op)
```

Comment:
Is there a reason that we need to look at source tables instead of the output schema of the immediate child? (i.e., op.table)